### PR TITLE
Shows membership information in wrangler whoami

### PR DESCRIPTION
**Pull request by danielrs**
## What this PR solves / how to test

This PR will show membership information in addition to token scopes when a request against an account-scoped resource fails with Authentication Error (code=10000). This is useful because sometimes requests will fail due to membership permissions, even if the token has the appropriate permissions.

For example, here's the last few lines of what we show now if I try to upload a Worker to an account I'm a member of with read-only permissions:

```
...
🔓 Token Permissions: If scopes are missing, you may need to logout and re-login.
Scope (Access)
- account (read)
- user (read)
- workers (write)
- workers_kv (write)
- workers_routes (write)
- workers_scripts (write)
- workers_tail (read)
- d1 (write)
- pages (write)
- zone (read)
- ssl_certs (write)
- ai (write)
- queues (write)
- offline_access
🎢 Membership roles in "Daniel Account": Contact account admin if you need more roles.
- Administrator Read Only
```

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: unit tests are enough.
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: not a new feature, just some extra information in `whoami`.

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
Nice PR!

**Comment by petebacondarwin** [comment]
Unfortunately CI doesn't agree. Need some work on the tests.

File: packages/wrangler/src/whoami.ts [hunk 1]
```diff
@@ -52,6 +61,40 @@ export async function whoami() {
...
+		);
+		if (!selectedAccount) {
+			return;
+		}
+		const membershipRoles = await fetchMembershipRoles(selectedAccount.id);
+		if (!membershipRoles) {
+			return;
+		}
+		logger.log(
+			`🎢 Membership roles in "${selectedAccount.name}": Contact account admin if you need more roles.`
```
Comment (JoshuaYe) [inline packages/wrangler/src/whoami.ts:81]: Copy suggestion: 🎢 Membership roles in "${selectedAccount.name}": Contact account admin if you lack permissions for your work.

**Review by JoshuaYe — CHANGES_REQUESTED** [review]
Made small copy suggestion, aside from that looks great Daniel :D

File: packages/wrangler/src/whoami.ts (hunk 1 above)
Comment (tanushree-sharma) [reply packages/wrangler/src/whoami.ts:81]: Membership roles in "${selectedAccount.name}": Contact your account's super admin to change your permissions.

**Pull request merged**